### ISSUE TITLE
QtWebEngine windows on unfocused monitors goes white (eg. qutebrowser)

### ISSUE THREAD

**Issue by JonathanReeve** [paperwm/PaperWM#270]
I can only make this happen with Qutebrowser, so maybe it's a problem with Qutebrowser? But then again, this only ever happens while using PaperWM, too. 

If I have Qutebrowser open and something else on another monitor, it's no problem, so long as Qutebrowser is focused. However, if I focus a window on another desktop, Qutebrowser stops displaying the web page and only displays a white screen. This is a problem if I want to have a web page open and also work on something else on a different monitor. 

I don't know if this is Qutebrowser's native behavior for some state of the window, or what. But I'm happy to help debug this however I can. 

Great extension, by the way. I'm loving this.

**Comment by JonathanReeve** [comment]
Just in case, I added a bug report [here at Qutebrowser's repo, too](https://github.com/qutebrowser/qutebrowser/issues/5288).

**Comment by olejorgenb** [comment]
Interesting (reproducible here)...

Due to how workspaces are implemented in gnome-shell/mutter[1] we show _clones_ and not the _actual_ windows on non-focused monitors. 

It seems like qutebrowser somehow listens to whether the window is visible and suspends rendering **and** clears most of the buffer. (Suspending rending kinda makes sense, but not sure why they would clear the buffer..)

The effect can also be observed by minimizing the window and using a native gnome-shell feature utilizing clones [2]. Eg. opening the gnome-shell overview or alt-tabbing. The qutebrowser window-preview is white. As such I would say this is a bug/misbehavior of qutebrowser.

[1] Only one workspace can be visible at a time. We want per-monitor workspaces so we "fake" it using window clones.
[2] I haven't tested without paperwm active, but we don't really interfere much with the overview.

**Comment by hedning** [comment]
Yeah, this probably isn't something we can fix in paperwm. We'd have to use single workspace for all windows and re-implement workspaces on top of that. We've considered doing that, but decided against as it would break all external tools etc. which relies on native workspaces.

**Comment by wolass** [comment]
Just for the purpose of finding this thread when people wonder why their RStudio window goes blank when switching workspaces. 

I hope that you could work out how to set up one workspace for multiple monitors as an alternative setting to multi-workspace environment.
 
I use mainly two apps for research (data science and academic writing) and avoid using the mouse:
- qutebrowser
- Rstudio

that means that I effectively have just one workspace if I put these on separate screens (making use of that real estate) and it kind of misses the point of having multiple monitors... 
Too bad that my two most used apps are affected by this problem...

**Comment by gerard67** [comment]
Hi, thanks for the report and the explanations.
Adding here to mention that the behavior is the same with kitty terminal and Intellij

**Comment by ssiegler** [comment]
In kitty and IntelliJ it is a little bit less severe. The windows on the inactive screens still display, but become stale and don't render updates. 

Firefox 102.0 shows even worse behavior: When switching to another monitor the Firefox window is not shown at all.

**Comment by Thesola10** [comment]
Superseded by #943 

---

_🧹 Just going through and triaging old issues..._
